### PR TITLE
Split hpp cpp

### PR DESCRIPTION
**Pull request by j042**
# Overview
This PR splits header files and non-template implementations. As a result, `cpp` folder is split into `include` and `src` folder.

## Addressed issues
* too many files in `cpp` 

## Note
Waiting for #194. Then py_multi_patch.hpp will be properly separated.

### CODE REVIEW

File: include/splinepy/fitting/lu_solve.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <cmath>
+#include <vector>
+
+namespace splinepy::fitting {
+
+void Doolittle(const std::vector<double>& matrix,
```
Comment (jzwar) [inline include/splinepy/fitting/lu_solve.hpp:8]: Maybe some comments? (I would also accept "next PR")

File: src/py/py_spline.cpp [hunk 2]
```diff
@@ -0,0 +1,927 @@
...
+            kv_dim,
+            ").",
+            "At least",
+            n_minimal_knots,
+            "knots are expected, but",
+            nknots,
+            "were given.");
+      }
+
+      // Only use closed knot-vectors
```
Comment (jzwar) [inline src/py/py_spline.cpp:115]:
```suggestion
```

**Review by jzwar — APPROVED** [review]
I guess that most of these changes are renamings and in some cases splitting of hpp and cpp. I did not read all the files in detail, so if there are any sneaky changes in the actual implementation, well played :wink: 
Thanks for putting in the effort. You are instantly rewarded with 6e4 line changes :D LGTM

File: include/splinepy/fitting/lu_solve.hpp (hunk 1 above)
Comment (j042) [reply include/splinepy/fitting/lu_solve.hpp:8]: next PR

**Pull request merged**